python/ray/data/block.py

```diff
@@ -174,6 +175,27 @@ def _apply_strict_mode_batch_format(given_batch_format: Optional[str]) -> str:
     return given_batch_format
 
 
+def _apply_strict_mode_batch_size(
+    given_batch_size: Optional[Union[int, Literal["default"]]], use_gpu: bool
+) -> Optional[int]:
+    ctx = ray.data.DatasetContext.get_current()
+    if ctx.strict_mode:
+        if use_gpu and not given_batch_size or given_batch_size == "default":
+            raise StrictModeError(
+                "`batch_size` must be provided to `map_batches` when requesting GPUs."
```

Comment:
Let's add a little more explanation here, something like
```
The optimal batch size for ML models depends on the model, data and GPU used -- try the setting that comes recommended with your model or example script, or try out which setting is the most performant for your situation and fits on GPU memory. Typically smaller batch sizes like 1, 2, 4, 8, 16, 32 are optimal for large models or memory constrained environments and larger batch sizes like 64, 128, 256 or 512 are optimal for smaller models or large GPUs.
```

Reply:
We could also link a doc page that goes a little more into this and e.g. shows people how to use the dashboard to see GPU memory utilization (possibly even link to a GPU memory profiler or integrate one into the dashboard if available).

Reply:
updated error message